BrainPortal/app/controllers/userfiles_controller.rb

```diff
@@ -791,19 +791,20 @@ def quality_control_panel #:nodoc:
     fail_tag    = current_user.available_tags.find_or_create_by_name_and_user_id_and_group_id("QC_FAIL", admin_user.id, everyone_group.id)
     unknown_tag = current_user.available_tags.find_or_create_by_name_and_user_id_and_group_id("QC_UNKNOWN", admin_user.id, everyone_group.id)
 
-    commit_name     = extract_params_key([ :next, :previous, :pass, :fail, :unknown ])
+    commit_name     = extract_params_key([ :next, :previous, :pass, :fail, :unknown, :update ])
     if @current_index >=  0 && commit_name && commit_name != :next && commit_name != :previous
       @current_userfile = Userfile.find_accessible_by_user(@filelist[@current_index], current_user, :access_requested => :read)
       tag_ids = params[:tag_ids] || []
       case commit_name
-      when :pass
-        tag_ids |= [pass_tag.id.to_s]
-      when :fail
-        tag_ids |= [fail_tag.id.to_s]
-      when :unknown
-        tag_ids |= [unknown_tag.id.to_s]
+        when :pass
+          tag_ids |= [pass_tag.id.to_s]
+        when :fail
+          tag_ids |= [fail_tag.id.to_s]
+        when :unknown
+          tag_ids |= [unknown_tag.id.to_s]
       end
-      @current_userfile.set_tags_for_user(current_user, tag_ids)
+      @current_userfile.set_tags_for_user(current_user, tag_ids) if tag_ids
+      @current_userfile.update_attributes_with_logging(params[:userfile], current_user) if params[:userfile][:description]
```

Comment:
Doesn't that allow a user to inject and modify any other attribtues in the userfile at the same time? If I edit the HTML and add a tag such as:

```html
   <input type="hidden" name="userfiles[name]" value="abcd">
```

wouldn't that change the name? Woudlnt' the same trick allow someone to chnage critical security-related attributes, like group_id, read_only, etc etc ?